.github/workflows/codeql-analysis.yml

```diff
@@ -1,76 +1,37 @@
-# For most projects, this workflow file will not need changing; you simply need
-# to commit it to your repository.
-#
-# You may wish to alter this file to override the set of languages analyzed,
-# or to provide custom queries or build logic.
-#
-# ******** NOTE ********
-# We have attempted to detect the languages in your repository. Please check
-# the `language` matrix defined below to confirm you have the correct set of
-# supported CodeQL languages.
-#
-name: "CodeQL"
-
+name: 'CodeQL'
 on:
-  push:
-    branches: [ main ]
   pull_request:
-    # The branches below must be a subset of the branches above
-    branches: [ main ]
+    types: [ opened, reopened, synchronize ]
+    paths:
+      - 'src/main/java/**'
   schedule:
```

Comment:
Do we need this if we are doing it on a pull_request?

Reply:
the schedule? its for services that aren't under much active development so we still get new alerts

Reply:
Yeah I didn't think there would be any new alerts - but I suppose its better than not having it.